geode-core/src/main/java/org/apache/geode/internal/SystemTimer.java

```diff
@@ -181,88 +167,67 @@ private static void sweepAllSwarms() {
   /**
    * Remove given timer from the swarm.
    *
-   * @param t timer to remove
+   * @param timerToRemove timer to remove
    *
    * @see #cancel()
    */
-  private static void removeFromSwarm(SystemTimer t) {
-    final boolean isDebugEnabled = logger.isTraceEnabled();
+  private static void removeFromSwarm(SystemTimer timerToRemove) {
     synchronized (allSwarms) {
       // Get timer's swarm
-      ArrayList swarmSet = (ArrayList) allSwarms.get(t.swarm);
+      List<WeakReference<SystemTimer>> swarmSet = allSwarms.get(timerToRemove.swarm);
       if (swarmSet == null) {
-        if (isDebugEnabled) {
-          logger.trace("SystemTimer#removeFromSwarm: timer already removed: {}", t);
-        }
         return; // already gone
       }
 
       // Remove timer from swarm
-      if (isDebugEnabled) {
-        logger.trace("SystemTimer#removeFromSwarm: removing timer <{}>", t);
-      }
       synchronized (swarmSet) {
-        Iterator it = swarmSet.iterator();
-        while (it.hasNext()) {
-          WeakReference ref = (WeakReference) it.next();
-          SystemTimer t2 = (SystemTimer) ref.get();
-          if (t2 == null) {
-            // Since we've discovered an empty reference, we should remove it.
-            it.remove();
-            continue;
-          }
-          if (t2 == t) {
-            it.remove();
+        Iterator<WeakReference<SystemTimer>> swarmIterator = swarmSet.iterator();
+        while (swarmIterator.hasNext()) {
+          WeakReference<SystemTimer> ref = swarmIterator.next();
+          SystemTimer timer = ref.get();
+          if (timer == null || timer == timerToRemove) {
+            swarmIterator.remove();
```

Comment:
no more `continue`...

Reply:
_continue_ isn't needed - this is the end of the loop